ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/common/FailureXAResource.java

```diff
@@ -9,146 +9,166 @@
 import javax.transaction.xa.XAResource;
 import javax.transaction.xa.Xid;
 
-public class FailureXAResource implements XAResource
-{
-    public enum FailLocation { none, prepare, commit, rollback, end, prepare_and_rollback };
-    public enum FailType { normal, timeout, heurcom, nota, inval, proto, rmfail, rollback, XA_RBCOMMFAIL, XA_HEURHAZ, message };
-    
-    public FailureXAResource ()
-    {
+public class FailureXAResource implements XAResource {
+    public enum FailLocation {
+        none,
+        prepare,
+        commit,
+        rollback,
+        end,
+        prepare_and_rollback
+    }
+
+    public enum FailType {
+        normal,
+        timeout,
+        heurcom,
+        nota,
+        inval,
+        proto,
+        rmfail,
+        rollback,
+        XA_RBCOMMFAIL,
+        XA_HEURHAZ,
+        message,
+        XA_RBINTEGRITY
+    }
+
+    public FailureXAResource() {
         this(FailLocation.none, FailType.normal);
     }
-    
-    public FailureXAResource (FailLocation loc)
-    {
+
+    public FailureXAResource(FailLocation loc) {
         this(loc, FailType.normal);
     }
-    
-    public FailureXAResource (FailLocation loc, FailType type)
-    {
+
+    public FailureXAResource(FailLocation loc, FailType type) {
         _locale = loc;
         _type = type;
     }
-    
-    public void commit(Xid id, boolean onePhase) throws XAException
-    {
-	if (_locale == FailLocation.commit)
-	{
-	    if (_type == FailType.normal)
-	        throw new XAException(XAException.XA_HEURMIX);
-	    
-	    if (_type == FailType.heurcom)
-	        throw new XAException(XAException.XA_HEURCOM);
-	    
-	    if (_type == FailType.rollback)
-                throw new XAException(XAException.XA_HEURRB);
-	    
-	    if (_type == FailType.nota)
-	        throw new XAException(XAException.XAER_NOTA);
-	    
-	    if (_type == FailType.inval)
-                throw new XAException(XAException.XAER_INVAL);
-	    
-	    if (_type == FailType.proto)
-                throw new XAException(XAException.XAER_PROTO);
-	    
-	    if (_type == FailType.rmfail)
-                throw new XAException(XAException.XAER_RMFAIL);
-	    
-	    throw new XAException(XAException.XA_RBTIMEOUT);
-	}
-    }
-
-    public void end(Xid xid, int flags) throws XAException
-    {
-        if (_locale == FailLocation.end)
-        {
-            if (_type == FailType.normal)
-                throw new XAException(XAException.XA_HEURRB);
-            
-            if (_type == FailType.timeout)
-                throw new XAException(XAException.XA_RBTIMEOUT);
-            
-            if (_type == FailType.XA_RBCOMMFAIL)
-                throw new XAException(XAException.XA_RBCOMMFAIL);
+
+    public void commit(Xid id, boolean onePhase) throws XAException {
+        if (_locale == FailLocation.commit) {
+            switch (_type) {
```

Comment:
What prompted the change from the if (_type... to a switch style of comparison?

Reply:
I preferred the switch style because it fits better this kind of logic (i.e. the `switch` statement can have a number of possible execution paths...which here was simulated with a sequence of `if` statements)

Reply:
I will add the note here that I don't know if I think that the rule we have (https://github.com/jbosstm/narayana/blob/main/CONTRIBUTING.md#coding-guidelines) permits the change from `if, then, else` to `switch`.

Reply:
I rolled back this style modification.